spring-cloud-config-server/src/test/java/org/springframework/cloud/config/server/environment/VaultEnvironmentRepositoryTests.java

```diff
@@ -55,6 +54,89 @@ public void testFindOne() throws IOException {
 		assertEquals(result, e.getPropertySources().get(0).getSource());
 	}
 
+	@Test
+	public void testFindOne_default_key_set_and_different_to_application() throws IOException {
+		MockHttpServletRequest configRequest = new MockHttpServletRequest();
+		configRequest.addHeader("X-CONFIG-TOKEN", "mytoken");
+		RestTemplate rest = Mockito.mock(RestTemplate.class);
+
+		ResponseEntity<VaultEnvironmentRepository.VaultResponse> myAppResp = Mockito.mock(ResponseEntity.class);
+		Mockito.when(myAppResp.getStatusCode()).thenReturn(HttpStatus.OK);
+		VaultEnvironmentRepository.VaultResponse myAppVaultResp = Mockito.mock(VaultEnvironmentRepository.VaultResponse.class);
+		Mockito.when(myAppVaultResp.getData()).thenReturn("{\"foo\":\"bar\"}");
+		Mockito.when(myAppResp.getBody()).thenReturn(myAppVaultResp);
+		Mockito.when(rest.exchange(Mockito.eq("http://127.0.0.1:8200/v1/{backend}/{key}"),
+				Mockito.eq(HttpMethod.GET), Mockito.any(HttpEntity.class), Mockito.eq(VaultEnvironmentRepository.VaultResponse.class),
+				Mockito.eq("secret"), Mockito.eq("myapp"))).thenReturn(myAppResp);
+
+		ResponseEntity<VaultEnvironmentRepository.VaultResponse> myDefaultKeyResp = Mockito.mock(ResponseEntity.class);
+		Mockito.when(myDefaultKeyResp.getStatusCode()).thenReturn(HttpStatus.OK);
+		VaultEnvironmentRepository.VaultResponse myDefaultKeyVaultResp = Mockito.mock(VaultEnvironmentRepository.VaultResponse.class);
+		Mockito.when(myDefaultKeyVaultResp.getData()).thenReturn("{\"def-foo\":\"def-bar\"}");
+		Mockito.when(myDefaultKeyResp.getBody()).thenReturn(myDefaultKeyVaultResp);
+		Mockito.when(rest.exchange(Mockito.eq("http://127.0.0.1:8200/v1/{backend}/{key}"),
+				Mockito.eq(HttpMethod.GET), Mockito.any(HttpEntity.class), Mockito.eq(VaultEnvironmentRepository.VaultResponse.class),
+				Mockito.eq("secret"), Mockito.eq("mydefaultkey"))).thenReturn(myDefaultKeyResp);
+
+		ResponseEntity<VaultEnvironmentRepository.VaultResponse> appResp = Mockito.mock(ResponseEntity.class);
```

Comment:
Awesome tests btw, though I don't think you need to mock `VaultResponse` or `ResponseEntity`.